src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.ts

```diff
@@ -0,0 +1,164 @@
+import { MatDialog } from '@angular/material/dialog';
+import { ProjectType } from 'realtime-server/lib/esm/scriptureforge/models/translate-config';
+import { combineLatest, of, Subscription } from 'rxjs';
+import { switchMap, tap } from 'rxjs/operators';
+import { BuildDto } from 'src/app/machine-api/build-dto';
+import { ActivatedProjectService } from 'xforge-common/activated-project.service';
+import { DialogService } from 'xforge-common/dialog.service';
+import { I18nService } from 'xforge-common/i18n.service';
+import { Locale } from 'xforge-common/models/i18n-locale';
+import { SubscriptionDisposable } from 'xforge-common/subscription-disposable';
+import { BuildStates } from '../../machine-api/build-states';
+import { NllbLanguageService } from '../nllb-language.service';
+import { ACTIVE_BUILD_STATES } from './draft-generation';
+import { DraftGenerationService } from './draft-generation.service';
+
+@Component({
+  selector: 'app-draft-generation',
+  templateUrl: './draft-generation.component.html',
+  styleUrls: ['./draft-generation.component.scss']
+})
+export class DraftGenerationComponent extends SubscriptionDisposable implements OnInit {
+  draftJob?: BuildDto;
+  draftViewerUrl?: string;
+
+  targetLanguage?: string;
+  targetLanguageDisplayName?: string;
+
+  isTargetLanguageNllb = true;
+  isBackTranslation = true;
+
+  jobSubscription?: Subscription;
+  isReady = false;
+
+  constructor(
+    private readonly matDialog: MatDialog,
+    private readonly dialogService: DialogService,
+    public readonly activatedProject: ActivatedProjectService,
+    private readonly draftGenerationService: DraftGenerationService,
+    private readonly nllbService: NllbLanguageService,
+    private readonly i18n: I18nService,
+    @Inject(ACTIVE_BUILD_STATES) private readonly activeBuildStates: BuildStates[]
+  ) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.subscribe(
+      combineLatest([this.activatedProject.projectId$, this.activatedProject.projectDoc$, this.i18n.locale$]),
+      ([projectId, projectDoc, locale]) => {
+        this.isBackTranslation = projectDoc?.data?.translateConfig.projectType === ProjectType.BackTranslation;
+        this.targetLanguage = projectDoc?.data?.writingSystem.tag;
+        this.targetLanguageDisplayName = this.getLanguageDisplayName(this.targetLanguage, locale);
+        this.isTargetLanguageNllb = this.nllbService.isNllbLanguage(this.targetLanguage);
+        this.draftViewerUrl = `/projects/${projectId}/draft-preview`;
+      }
+    );
+
+    this.jobSubscription = this.subscribe(
+      this.draftGenerationService
+        .getBuildProgress(this.activatedProject.projectId!)
+        .pipe(
+          switchMap((job?: BuildDto) =>
+            this.isDraftInProgress(job)
+              ? this.draftGenerationService.pollBuildProgress(this.activatedProject.projectId!)
+              : of(job)
+          )
+        ),
+      job => {
+        this.draftJob = job;
```

Comment:
I notice you use `draftJob` to determine whether or not a draft has been generated (and so ready for the next step).

When a build has finished, `getBuildProgress` will return a 404 - as there is not build running.

Is there another way we could calculate the "build completed" state? If there are API changes you want me to make, I can do them.

Also, I have noticed an error in my API - MachineApService.cs line 816 should be:
```cs
            case ServalApiException { StatusCode: StatusCodes.Status204NoContent }:
                return;
```

The machine-api should be telling you a 204 if there is no build running.